dplutils/pipeline/stream.py

```diff
@@ -173,33 +175,39 @@ def _handle_one_task(task, rank):
                 for batch in deque_extract(task.data_in, lambda b: b.length > batch_size):
                     task.split_pending.appendleft(self.split_batch_submit(batch, batch_size))
 
-            while len(task.data_in) > 0:
-                num_to_merge = deque_num_merge(task.data_in, batch_size)
-                if num_to_merge == 0:
-                    # If the feed is terminated and there are no more tasks that
-                    # will feed to this one, submit everything
-                    if self.source_exhausted and self.task_exhausted(task):
-                        num_to_merge = len(task.data_in)
-                    else:
-                        break
-                eligible = True
-                if not self.task_submittable(task.task, rank):
-                    break
-                merged = [task.data_in.pop().data for i in range(num_to_merge)]
-                task.pending.appendleft(self.task_submit(task.task, merged))
-                task.counter += 1
-                submitted = True
+            num_to_merge = deque_num_merge(task.data_in, batch_size)
```

Comment:
fyi: the change in this block simply removes the for loop, submitting (if eligible) only a single invocation of task at a time (to enable the fairness re-evaluation). The `break` -> `return` makes the diff look like more than whitespace.